x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/authorization/ElasticInferenceServiceAuthorization.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.elastic.authorization;
+
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.xpack.inference.external.response.elastic.ElasticInferenceServiceAuthorizationResponseEntity;
+
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Provides a structure for governing which models (if any) a cluster has access to according to the upstream Elastic Inference Service.
+ * @param enabledModels a mapping of model ids to a set of {@link TaskType} to indicate which models are available and for which task types
+ */
+public record ElasticInferenceServiceAuthorization(Map<String, EnumSet<TaskType>> enabledModels) {
+
+    /**
+     * Converts an authorization response from EIS into the {@link ElasticInferenceServiceAuthorization} format.
```

Comment:
```suggestion
     * Converts an authorization response from Elastic Inference Service into the {@link ElasticInferenceServiceAuthorization} format.
```

nit: I think we don't want to use `EIS` in any docs or class names